src/sections/pools/modals/AddLiquidity/AddLiquidity.utils.ts

```diff
@@ -388,3 +395,40 @@ export const useXYKZodSchema = (
 
 export const getXYKPoolShare = (total: BigNumber, add: BigNumber) =>
   add.div(total.plus(add)).multipliedBy(100)
+
+export const useXYKPoolJoinFarmMinShares = (
+  poolAddress: string,
+  minDeposit: BN,
+) => {
+  const { data: xykPools } = useXYKSDKPools()
+
+  const [assetAReserve, assetBReserve] =
+    xykPools?.find((xykPool) => xykPool.address === poolAddress)?.tokens ?? []
+
+  const { data: oracle } = useOraclePrice(
+    assetAReserve?.id,
+    assetBReserve?.id,
+    "hydraxyk",
+  )
+
+  return useMemo(() => {
+    const sharesNew = calculate_liquidity_out_asset_a(
```

Comment:
can you return undefined or something if oracle is undefined instead of passing 0 to each argument